randomizers/items.py

```diff
@@ -452,9 +452,6 @@ def change_chest_item(self, arc_path: str, chest_index: int, layer: DZxLayer, it
       # Here, we set that custom trap flag.
       chest.behavior_type |= 0x40
 
-      if self.options.get("chest_type_matches_contents"):
-        chest.chest_type = self.rng.randrange(0, 2)
```

Comment:
When CTMC is on, will trap chests be the same type as the chest in the vanilla game? That seems a bit strange

Reply:
Yes, that is the case. I had to remove the randomization here because `self.rng` is invalid at this phase of the process. @LagoLunatic would have to ensure that the object remains valid for me to use it, but they expressed concerns that having access to the RNG object in the saving phase could cause problems.